modules/apps/remote-web-app/remote-web-app-web/src/main/java/com/liferay/remote/web/app/web/internal/display/context/RemoteWebAppAdminViewManagementToolbarDisplayContext.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.remote.web.app.web.internal.display.context;
+
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.CreationMenu;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.CreationMenuBuilder;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemListBuilder;
+import com.liferay.portal.kernel.dao.search.EmptyOnClickRowChecker;
+import com.liferay.portal.kernel.dao.search.SearchContainer;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.PortletURLUtil;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.remote.web.app.model.RemoteWebAppEntry;
+import com.liferay.remote.web.app.service.RemoteWebAppEntryLocalService;
+
+import java.util.List;
+
+import javax.portlet.PortletURL;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class RemoteWebAppAdminViewManagementToolbarDisplayContext {
+
+	public RemoteWebAppAdminViewManagementToolbarDisplayContext(
+		RenderRequest renderRequest, RenderResponse renderResponse,
+		RemoteWebAppEntryLocalService remoteWebAppEntryLocalService) {
+
+		_renderRequest = renderRequest;
+		_renderResponse = renderResponse;
+		_remoteWebAppEntryLocalService = remoteWebAppEntryLocalService;
+
+		_searchContainer = new SearchContainer<>(
+			renderRequest, null, null, SearchContainer.DEFAULT_CUR_PARAM,
+			SearchContainer.DEFAULT_DELTA, getCurrentPortletURL(), null,
+			"no-entries-were-found");
+
+		_searchContainer.setRowChecker(
+			new EmptyOnClickRowChecker(renderResponse));
+
+		List<RemoteWebAppEntry> remoteWebAppEntries =
+			_remoteWebAppEntryLocalService.getRemoteWebAppEntries(0, 1000);
```

Comment:
This should say `QueryUtil.ALL_POS` instead of `1000` but anyway it's WIP, so it would probably be rewritten in the final version.

Reply:
It is WIP because the search has not been implemented...